app/services/set_public_id_for_qualification.rb

```diff
@@ -0,0 +1,34 @@
+class SetPublicIdForQualification
+
+  def initialize(qualification:)
+    @qualification = qualification
+  end
+
+  def call(preferred_id_value: nil)
+    top_level_public_id = @qualification.public_id.presence || preferred_id_value.presence || next_available_id
```

Comment:
This line means:
- If a public_id has already been set, don't touch it – this ensures consistency for the API
- We can set a value for the ID manually (as the rake task does, so we can enforce the id is the same in the API after the change)
- Otherwise, generate a new unique ID